oscduplicator/duplicator.py

```diff
@@ -0,0 +1,203 @@
+import json
+import queue
+from dataclasses import dataclass
+from threading import Thread
+from typing import Optional
+
+from pythonosc import dispatcher, osc_server, udp_client
+
+
+@dataclass
+class TransmitPortSetting:
+    """
+    再送信先のポートのデータを保持する
+
+    Parameters
+    ---------
+    name: str
+        ポートの名称
+    port: int
+        ポート番号
+    enable: bool
+        そのポートに再送信するかどうか
+    """
+
+    name: str
+    port: int
+    enabled: bool
+
+    def __post_init__(self):
+        # Validate the 'name' attribute
+        if not isinstance(self.name, str):
+            raise TypeError(
+                "Expected instance of type 'str' for attribute 'name',"
+                f"but got '{type(self.name).__name__}'."
+            )
+
+        # Validate the 'port' attribute
+        if not isinstance(self.port, int):
+            raise TypeError(
+                "Expected instance of type 'int' for attribute 'port',"
+                f" but got '{type(self.port).__name__}'."
+            )
+        if not (0 <= self.port <= 65535):  # standard port range for TCP/UDP
+            raise ValueError(
+                "Invalid 'port' value. Expected a number between 0 and 65535, "
+                f"but got '{self.port}'."
+            )
+
+        # Validate the 'enabled' attribute
+        if not isinstance(self.enabled, bool):
+            raise TypeError(
+                "Expected instance of type 'bool' for attribute 'enabled',"
+                f" but got '{type(self.enabled).__name__}'."
+            )
+
+
+class OSCDuplicator:
+    """
+    OSC信号を受信するためのサーバーと、再送信のための設定を保持する
+
+    Attributes
+    ---------
+    server: osc_server.ThreadingOSCUDPServer
+        OSC信号を受信するためのサーバー
+    receive_port: int
+        OSC信号を受信するためのポート、デフォルト9001
+    transmit_port_settings: list[TransmitPortSetting]
+        転送先ポートの設定を保持するリスト
+    clients: list[udp_client.SimpleUDPClient]
+        OSC信号を再送信するための、clientのリスト
+    q: queue.Queue
+        OSC信号の受信順・送信順を保証するためのキュー
+    """
+
+    def __init__(self) -> None:
+        self.__receive_port: Optional[int] = None
```

Comment:
```suggestion
        self.__receive_port: int | None = None
```

Python 3.10以降なので、`Optional` を使わなくてもよさそう
ref : https://docs.python.org/ja/3/library/typing.html?highlight=typing#typing.Optional